tutorials/FSS.ipynb

```diff
@@ -0,0 +1,2329 @@
+       "        [0.99975721, 0.81818033, 0.99351962],\n",
+       "        [0.99966944, 0.74355962, 0.96581436],\n",
+       "        [0.99942939, 0.66246222, 0.90830549],\n",
+       "        [0.99928144, 0.57516051, 0.81375132]],\n",
+       "\n",
+       "       [[0.99990218, 0.87900485, 0.99991912],\n",
+       "        [0.99984607, 0.81846299, 0.99364039],\n",
+       "        [0.99980318, 0.74363044, 0.96587942],\n",
+       "        [0.9996493 , 0.66234939, 0.90827159],\n",
+       "        [0.99954734, 0.5753859 , 0.81391288]]])\n",
+       "Coordinates:\n",
+       "  * window_size  (window_size) <U9 180B '[20 20]' '[40 40]' ... '[100 100]'\n",
+       "  * threshold    (threshold) <U4 80B '0.0' '0.25' '0.5' '0.75' '1.0'\n",
+       "  * scenario     (scenario) int64 24B 0 1 2\n",
+       "Attributes:\n",
+       "    description:  Fractions skill score"
+      ]
+     },
+     "execution_count": 4,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import xarray as xr\n",
+    "import itertools\n",
+    "import numpy as np\n",
+    "\n",
+    "# as before\n",
+    "window_sizes = np.linspace((20,20), (100,100), 5, dtype=(int,int))\n",
+    "thresholds = np.linspace(0.0, 1.0, 5, dtype=float)\n",
+    "input_scenarios = [[obs, fcst_1], [obs, fcst_2], [obs, fcst_3]]\n",
+    "compute_method = FssComputeMethod.NUMPY\n",
+    "\n",
+    "# create output array\n",
+    "W = len(window_sizes)\n",
+    "T = len(thresholds)\n",
+    "S = len(input_scenarios)\n",
+    "fss_out = np.zeros((W, T, S))\n",
+    "\n",
+    "# lets iterate over the indices, setting the `multi_index` and `writeonly` flags\n",
+    "with np.nditer(fss_out, flags=['multi_index'], op_flags=['writeonly']) as it:\n",
+    "    for _fss in it:\n",
+    "        # gather argument indices\n",
+    "        window_idx, threshold_idx, input_idx = it.multi_index\n",
+    "        _window_size = window_sizes[window_idx]\n",
+    "        _threshold = thresholds[threshold_idx]\n",
+    "        _obs, _fcst = input_scenarios[input_idx]\n",
+    "        # compute the fss for each combination of arguments\n",
+    "        _fss[...] = fss_2d_single_field(\n",
+    "            _fcst,\n",
+    "            _obs,\n",
+    "            event_threshold=_threshold,\n",
+    "            window_size=_window_size,\n",
+    "            compute_method=compute_method\n",
+    "        )\n",
+    "\n",
+    "# construct output xarray with results\n",
+    "da = xr.DataArray(\n",
+    "    data=fss_out,\n",
+    "    dims=[\"window_size\", \"threshold\", \"scenario\"],\n",
+    "    coords=dict(\n",
+    "        window_size=[str(x) for x in window_sizes],\n",
+    "        threshold=[str(x) for x in thresholds],\n",
+    "        scenario=range(len(input_scenarios))\n",
+    "    ),\n",
+    "    attrs=dict(\n",
+    "        description=\"Fractions skill score\",\n",
+    "    ),\n",
+    ")\n",
+    "da"
```

Comment:
Makes sense to me, it just starts to slow down the notebook if we increase the window size (since it's also part of the CI). I'll have a think about it but I'm gearing towards adding it in when some computational uplift (numba) gets added in.

Reply:
Similar to above, will resolve in separate issue when the multi window/threshold implementation is more well thought out.